include/datadog/baggage.h

```diff
@@ -0,0 +1,143 @@
+#include <datadog/dict_reader.h>
+#include <datadog/dict_writer.h>
+#include <datadog/expected.h>
+#include <datadog/optional.h>
+#include <datadog/string_view.h>
+
+#include <initializer_list>
+#include <string>
+#include <unordered_map>
+
+namespace datadog {
+namespace tracing {
+
+/// OpenTelemetry-like implementation of the Baggage concept.
+/// Baggage is a key-value store meant to propagate data across services and
+/// processes boundaries.
+///
+/// Baggage are extracted from any tracing context implementing the `DictReader`
+/// interface using `Baggage::extract`.
+///
+/// Baggages are injected to any tracing context implementing the `DictWriter`
+/// interface using the `inject` method.
+class Baggage {
+ public:
```

Comment:
The public API seems a little too large to me. Comparing this to the https://github.com/open-telemetry/opentelemetry-cpp/blob/5bc053e52280281736a4a07aec73d4f5f3500425/api/include/opentelemetry/baggage/baggage.h, can we remove the following API's from public accessibility:
- `contains`
- `size`
- `empty`
- `visit`

Reply:
Is there a specific reason we should limit ourselves to OpenTelemetry's API? Just for context, `dd-trace-cpp` isn’t designed to be OpenTelemetry-compatible and doesn’t aim to be. Ultimately, if a user wants to stay vendor-neutral, they’ll likely choose 🚧 `dd-opentelemetry-cpp` 🚧.

Reply:
There's two reasons why I think we should generally strive to match the OpenTelemetry API:

1. If our users may switch from using the OpenTelemetry API to the API in our library (I guess maybe not the case for this repo), then closely matching the structure can help make it easier for users to adopt our library.
2. In general I think we should try to provide the most minimal API we can (with the ability to add to it later as needed), and the OpenTelemetry API's have been discussed enough that they should be a good starting point for us to reference as well.

So if we can get by without the additional class members public, I would greatly prefer that. But if you think they are necessary then we can keep them, it's not a blocker.